coil-compose-benchmark/src/main/java/coil/compose/benchmark/BaselineProfileGenerator.kt

```diff
@@ -0,0 +1,25 @@
+package coil.compose.benchmark
+
+import androidx.benchmark.macro.ExperimentalBaselineProfilesApi
+import androidx.benchmark.macro.junit4.BaselineProfileRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@ExperimentalBaselineProfilesApi
+@RunWith(AndroidJUnit4::class)
+class BaselineProfileGenerator {
```

Comment:
How do we use this to determine if our baseline profile is effective? Is this something we can automate?

Reply:
This would be responsible for only generating the profile. There is also `BaselineProfileBenchmark` which is for determining if the profile is effective or not. Unfortunately, this would be big task and perhaps requires a GitHub discussion. Google has some plans to support baseline profile for CI/CD but nothing solid as of now.

Reply:
You can evaluate the effectiveness using the various CompilationMode flags in a macrobenchmark.  An example:

https://github.com/android/performance-samples/blob/main/MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/startup/StartupCompilationModeBenchmark.kt

They are actually running it against Firebase Test Lab devices for accuracy.

https://github.com/android/performance-samples/blob/main/.github/workflows/firebase_test_lab.yml#L72